pkg/controller/container-runtime-config/helpers.go

```diff
@@ -954,3 +955,92 @@ func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature
 	}
 	return data, nil
 }
+
+func addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig map[string]dockerConfig, scope string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy, idmsRules []*apicfgv1.ImageDigestMirrorSet, itmsRules []*apicfgv1.ImageTagMirrorSet, sigstoreAttachment dockerConfig) {
+	type sourceMirrors struct {
+		source  string
+		mirrors []string
+	}
+
+	processRulesElement := func(rule []sourceMirrors) bool {
+		scopeIsSource := false
+		for _, r := range rule {
+			if strings.HasPrefix(r.source, scope) {
```

Comment:
All of these `HasPrefix` match too much (`example.com/foo` matches `example.com/foobar`) _and_ too little (`*.example` does not match `registry.example`).

I’d expect (the semantics of) `ScopeIsNestedInsideScope` to be used.